server/src/FileChangeListener.ts

```diff
@@ -38,28 +44,38 @@ export class FileChangeListener {
   }
 
   onDidChangeWatchedFiles(params: DidChangeWatchedFilesParams): void {
-    const dbtProjectYmlPath = path.resolve(this.workspaceFolder, DbtRepository.DBT_PROJECT_FILE_NAME);
     const manifestJsonPath = path.resolve(this.workspaceFolder, this.dbtRepository.dbtTargetPath, DbtRepository.DBT_MANIFEST_FILE_NAME);
-    const packagesYmlPath = path.resolve(this.workspaceFolder, DbtRepository.DBT_PACKAGES_FILE_NAME);
     const packagesPaths = this.dbtRepository.packagesInstallPaths.map(p => path.resolve(this.workspaceFolder, p));
 
+    // For some paths we want to do action only once even we got several changes here
+    if (this.containsChangeWithPath(params.changes, manifestJsonPath)) {
+      this.updateManifestNodes();
+    }
```

Comment:
If we can use the same logic for `packagesPaths` ?

Reply:
We can. I did it. And not sure now, do we need debounce logic?